rules/TypeDeclaration/TypeInferer/AssignToPropertyTypeInferer.php

```diff
@@ -178,4 +145,76 @@ private function shouldAddNullType(ClassLike $classLike, string $propertyName, a
 
         return ! $hasPropertyDefaultValue;
     }
+
+    private function hasAssignDynamicPropertyValue(ClassLike $classLike, string $propertyName): bool
+    {
+        $hasAssignDynamicPropertyValue = false;
+
+        $this->simpleCallableNodeTraverser->traverseNodesWithCallable($classLike->stmts, function (Node $node) use (
+            $propertyName,
+            &$hasAssignDynamicPropertyValue,
+        ): ?int {
```

Comment:
inner class and inner function should by passed

```php
            if ($node instanceof Class_ || $node instanceof Function_) {
                // skip anonymous classes and inner function
                return NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN;
            }
```

Reply:
This is existing code which I just had to refactor because the complexity of `inferPropertyInClassLike` was getting to high. See the chang in this [commit](c97dc17079a970b6a6fda7682271b73304801f46).

Because it is not directly related to the natively typed properties this PR is about, I think it belongs in a new PR.  But I will add it to this PR if you would like me to, you may decide 😉.